client/utils/checkout.js

```diff
@@ -15,3 +15,22 @@ export const getConfig = ( name ) => {
 
 	return config[ name ] || null;
 };
+
+/**
+ * Forms dynamic gateway title for UPE checkout from enabled methods
+ *
+ * @param {Object} paymentMethodsConfig Object containing map of enabled UPE payment methods to settings.
+ * @return {string} Dynamic title string dependent on payment methods enabled.
+ */
+export const getCustomGatewayTitle = ( paymentMethodsConfig ) => {
+	const enabledPaymentMethods = Object.keys( paymentMethodsConfig ).sort();
+	let label = '';
+
+	if ( 2 > enabledPaymentMethods.length ) {
+		label = paymentMethodsConfig[ enabledPaymentMethods[ 0 ] ].title;
+	} else {
+		label = 'Popular payment methods';
```

Comment:
Could this be pulled from a config (or another solution) to allow for translation?

Reply:
Great idea! I moved this to a member variable on the gateway class where it can be translated and passed it in via the config, as you suggested. I think it's making the config a little crowded, but I can't really think of a better place to put it.